crates/websocket/src/websocket_listener.rs

```diff
@@ -81,9 +81,10 @@ async fn connect(
 ) -> Result<(WebsocketSender, WebsocketReceiver)> {
     match socket_result {
         Ok(socket) => {
-            socket.set_keepalive(Some(std::time::Duration::from_secs(
-                config.tcp_keepalive_s as u64,
-            )))?;
+            // TODO(steveeJ): investigate whether we need an alternative
+            // socket.set_keepalive(Some(std::time::Duration::from_secs(
+            //     config.tcp_keepalive_s as u64,
+            // )))?;
```

Comment:
if we really need to, we can use net2 https://docs.rs/net2/0.2.35/net2/trait.TcpStreamExt.html#tymethod.set_keepalive_ms to build a std tcp socket, then wrap it in tokio with https://docs.rs/tokio/0.3.3/tokio/net/struct.TcpStream.html#method.from_std

Reply:
Though, I guess that won't work for the incoming side...